configure.ac

```diff
@@ -851,25 +861,23 @@ AS_CASE([$ocaml_cc_vendor],
     # TODO: see whether the code can be fixed to avoid -Wno-unused
     common_cflags="-O2 -fno-strict-aliasing -fwrapv -mms-bitfields"
     internal_cppflags='-D__USE_MINGW_ANSI_STDIO=0 -DUNICODE -D_UNICODE'
-    internal_cppflags="$internal_cppflags -DWINDOWS_UNICODE="
-    internal_cppflags="${internal_cppflags}\$(WINDOWS_UNICODE)"],
+    internal_cppflags="$internal_cppflags -DWINDOWS_UNICODE=$windows_unicode"],
   [mingw-*],
     [AC_MSG_ERROR([Unsupported C compiler for a MinGW-w64 build])],
   [msvc-0*|msvc-1[[0-8]]*|msvc-19[[012]]*|msvc-193[[0-7]]],
     # No C11 atomics support
     [AC_MSG_ERROR(m4_normalize([This version of MSVC is too old.
     Please use Visual Studio version 17.8 or above.]), 69)],
   [msvc-*],
-    [common_cflags="-nologo -O2 -Gy- -MD $cc_warnings"
+    [common_cflags='-nologo -O2 -Gy- -MD'
     common_cppflags="-D_CRT_SECURE_NO_DEPRECATE"
-    internal_cppflags='-DUNICODE -D_UNICODE'
-    internal_cflags='-experimental:c11atomics -std:c11'
-    CFLAGS="$CFLAGS -experimental:c11atomics -std:c11"
-    OCAML_CL_HAS_VOLATILE_METADATA
-    AS_IF([test "x$cl_has_volatile_metadata" = "xtrue"],
-          [internal_cflags="$internal_cflags -d2VolatileMetadata-"])
-    internal_cppflags="$internal_cppflags -DWINDOWS_UNICODE="
-    internal_cppflags="${internal_cppflags}\$(WINDOWS_UNICODE)"],
+    internal_cflags="$cc_warnings"
+    internal_cppflags="-DUNICODE -D_UNICODE -D_CRT_NONSTDC_NO_WARNINGS \
+-D_WINSOCK_DEPRECATED_NO_WARNINGS"
```

Comment:
Just to make sure I understand: is `_WINSOCK_DEPRECATED_NO_WARNINGS` needed for the build to succeed, or is it just to silence the warnings. If the eventual objective is to migrate to different functions that do not trigger the warning, we may __not__ want to silence them; what do you think?

Reply:
The define is _solely_ needed to silence the warnings. However as the warnings are turned into errors for us devs, it's then needed for the build to succeed… There's also the option of passing `-Wno-deprecated-declarations` to clang, but the `_WINSOCK_DEPRECATED_NO_WARNINGS` define has the advantage of also working on MSVC. clang warns by default of deprecated functions. We'd need to turn on MSVC warnings with perhaps a high level to trigger them, but that is too noisy currently (I'm somewhat [planning on fixing them](https://github.com/MisterDA/ocaml/pull/41)).
The objective is indeed to move to newer functions (you may peek at https://github.com/MisterDA/ocaml/pull/38). In the meantime, my opinion is that we can move forward using this macro.

Reply:
OK, makes sense, thanks.

Reply:
Are there sufficiently few places that it could be done with `#define _WINSOCK_DEPRECATED_NO_WARNINGS /* Remove when gethostbyname replaced with getaddrinfo */` in the files? (I'm not 100% convinced this would be better)